server/app_flight.js

```diff
@@ -1,82 +1,148 @@
+      "engine": "google_flights",
+      "hl": "en",
+      "gl": "us",
+      "departure_id": "",
+      "arrival_id": "",
+      "outbound_date": "",
+      "return_date": "",
   }
-
-  const result = await model.generateContent(prompt);
-  const response = await result.response;
-  const text = response.text();
-
-
-  let trimmed_json_string = text.substring(text.indexOf('{'), text.indexOf('}')+1);
-  let trimmed_json = JSON.parse(trimmed_json_string)
-  Object.keys(trimmed_json).forEach(key => params[key] = trimmed_json[key])
-
+  generated_params_string = generated_params_string['arguments']
+  let generated_params = JSON.parse(generated_params_string)
+  Object.keys(generated_params).forEach(key => params[key] = generated_params[key])
+  // console.log(params)
   return params
 }
 
-async function run(){
-  const initial_prompt = "Give me flight options from los angeles to jfk on feb 23 to feb 27 in 2024"
-  let error_msg = ""
+async function retrieve_flight_options(user_prompt) {
+  const initial_prompt = user_prompt;
+  let error_msg = "";
   let best_response = {}
 
   let i = 0
-  while(i < 5){
-    params = await gen_flight_params(initial_prompt, error_msg)
-    console.log(params)
-    //const response = await getJson(params);
-    try{
-      const response = await getJson(params);
-
-      if (response["error"]){
-        throw new Error("\n \n There is something wrong with the JSON you provided last time I made this query. Have you considered that the airport codes you are using don't actually exist? This has to be an airport code corresponding to exactly one airport, not a destination code that simply refers to a region with multiple airports.");
-      } 
-      best_response = response;
-      break;  
-    }
-    catch(e){
-      error_msg += e
-      console.log(e)
-    }
-    i++;
-  }
-  if (i==5){
-    //have some functionality to tell user to give a better prompt
-
+  while(i < 3){
+      try {
+          const generated_params = await create_flight_request_parameters(initial_prompt, error_msg);
+          params = generate_flight_request_params(generated_params)
+          console.log('final_params', params)
+    
+          const response = await getJson(params)
+          console.log(response)
+          if (response["error"]){
+              throw new Error("\n \n There is something wrong with the JSON you provided last time I made this query.");
+          } 
+          best_response = response;
+          break;
+
+      } catch(e) {
+          error_msg += e
+          console.error(e);
+      }
+      i++;
   }
-  
+  return best_response
+}
 
+async function return_flight_options() {
+  const initial_prompt = "Help me plan a trip for Korea. I have 3 triplets and will travel with my husband";
 
+  flight_retrieval_results = await retrieve_flight_options(initial_prompt);
+  console.log(flight_retrieval_results['best_flights']) 
+/** 
+  try {
```

Comment:
Let's do this map. Slice 0,5 best flights and then get the things we need

We need both airport codes, flight duration, flight take off time and flight land time, flight airline logo, num stops, and price, travel class